codalab/rest/schemas.py

```diff
@@ -155,11 +155,19 @@ class BundleDependencySchema(PlainSchema):
     parent_uuid = fields.String(validate=validate_uuid)
     parent_path = fields.String(missing="")
     parent_name = fields.Method('get_parent_name', dump_only=True)  # for convenience
+    parent_state = fields.Method('get_parent_state', dump_only=True)  # for convenience
 
     def get_parent_name(self, dep):
         uuid = dep['parent_uuid']
         return local.model.get_bundle_names([uuid]).get(uuid)
 
+    def get_parent_state(self, dep):
+        uuid = dep['parent_uuid']
+        try:
+            return local.model.get_bundle_state(uuid)
+        except NotFoundError:
```

Comment:
when would NotFoundError be thrown? Do we need this try-except?

Reply:
@epicfaace I observed an edge case in which `NotFoundError` is thrown [here](https://github.com/codalab/codalab-worksheets/blob/master/codalab/model/bundle_model.py#L1252) when we try to fetch a state for dependencies that were force-deleted.